### ISSUE TITLE
Does the KL divergence calculation correct?

### ISSUE THREAD

**Issue by drmeerkat** [pytorch/pytorch#43348]
## To Reproduce

`
import torch.nn.functional as F
a = torch.tensor([.5, .5])
print(F.kl_div(a, a, reduction='sum'))
`

## Expected behavior

I am expecting the output to be 0

## Environment

1.6.0 MacOS conda cpu

## Additional context

It seems that in file 

> pytorch/aten/src/ATen/native/Loss.cpp

`
Tensor kl_div(const Tensor& input, const Tensor& target, int64_t reduction) {
  auto zeros = at::zeros_like(target);
  auto output_pos = target * (at::log(target) - input);
  auto output = at::where(target > 0, output_pos, zeros);
  return apply_loss_reduction(output, reduction);
}
`
line 71 it should be log(input)

**Comment by SsnL** [comment]
> As with NLLLoss, the input given is expected to contain log-probabilities and is not restricted to a 2D Tensor. The targets are interpreted as probabilities by default, but could be considered as log-probabilities with log_target set to True.

https://pytorch.org/docs/stable/generated/torch.nn.KLDivLoss.html#torch.nn.KLDivLoss

**Comment by drmeerkat** [comment]
Yeah, I saw that line. But the document is quite confusing at first glance. Would it be better to make it more consistent with the original equations?

**Comment by SsnL** [comment]
It's unclear what you mean by "original equations", but it is exactly the same as the equations in the documentation though.

**Comment by drmeerkat** [comment]
Let me put this another way. I personally prefer the default value of ‘log_target’ to be true so that the input and the target value is in the same scale by default, which may reduce a lot of confusion.

**Comment by SsnL** [comment]
I didn't author the function, but the docs say that it is this way to be consistent with xce loss, which makes sense to me. I understand where your argument comes from, but it seems in this cases confusion may happen in either case. And I don't think this is worth doing a silent BC-breaking change.

**Comment by drmeerkat** [comment]
Gotcha, thanks!
